### PR TITLE
feat: added key generation command to EN

### PR DESCRIPTION
**Pull request by pompon0**
it will be used by partners running ENs to populate consensus secrets config. Also drafted the documentation on how to enable gossipnet on EN deployment. Following the instructions using a docker image won't be that simple though.

### CODE REVIEW

**Review by RomanBrodetski — APPROVED** [review]

**Review by aakoshh — DISMISSED** [review]

File: core/bin/external_node/src/main.rs [hunk 1]
```diff
@@ -778,9 +795,19 @@ async fn main() -> anyhow::Result<()> {
...
+            Command::GenerateSecrets => generate_consensus_secrets(),
+        }
+        return Ok(());
+    }
+
     let mut config = if let Some(config_path) = opt.config_path.clone() {
         let secrets_path = opt.secrets_path.clone().unwrap();
         let external_node_config_path = opt.external_node_config_path.clone().unwrap();
+        if opt.enable_consensus {
+            anyhow::ensure!(opt.consensus_path.is_some());
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/bin/external_node/src/main.rs]: done

File: core/bin/external_node/src/config/mod.rs [hunk 2]
```diff
@@ -1126,6 +1128,18 @@ impl ExperimentalENConfig {
     }
 }
 
+pub fn generate_consensus_secrets() {
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/bin/external_node/src/config/mod.rs:1134]: done

File: core/bin/external_node/src/config/mod.rs [hunk 3]
```diff
@@ -1126,6 +1128,18 @@ impl ExperimentalENConfig {
...
+    let validator_key = roles::validator::SecretKey::generate();
+    let attester_key = roles::attester::SecretKey::generate();
+    let node_key = roles::node::SecretKey::generate();
+    println!("# {}", validator_key.public().encode());
+    println!("- validator_key: '{}'", validator_key.encode());
+    println!("# {}", attester_key.public().encode());
+    println!("- attester_key: '{}'", attester_key.encode());
+    println!("# {}", node_key.public().encode());
+    println!("- node_key: '{}'", node_key.encode());
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/bin/external_node/src/config/mod.rs]: done

**Review by popzxc — DISMISSED** [review]

File: docs/guides/external-node/09_decentralization.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+> to the public internet, you can use IP in your local network.
+
+Currently the config contains the following fields (refer to config
+[schema](https://github.com/matter-labs/zksync-era/blob/990676c5f84afd2ff8cd337f495c82e8d1f305a4/core/lib/protobuf_config/src/proto/core/consensus.proto#L66)
+for more details):
+
+- `server_addr` - local TCP socket address that the node should listen on for incoming connections. Note that this is an
+  additional TCP port that will be opened by the node.
+- `public_addr` - the public address of your node that will be advertised over the gossipnet.
+- `max_payload_size` - limit (in bytes) on the sized of the ZKsync ERA block received from the gossipnet. This protects
```
Comment (aakoshh) [inline docs/guides/external-node/09_decentralization.md:55]: There is now a `max_batch_size` as well, which limits the response size for getting batches. I take it you don't want to mention it as it's not fully developed yet?

File: docs/guides/external-node/prepared_configs/mainnet_consensus_config.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+server_addr: '0.0.0.0:3054'
+public_addr: '<your public IP goes here>:3054'
+max_payload_size: 5000000
```
Comment (aakoshh) [inline docs/guides/external-node/prepared_configs/mainnet_consensus_config.yaml:3]: Any guess for a good `max_batch_size`? As per Roman's advise the zksync-era node will multiply it by 5000 if it's missing, which would be huge.

**Review by aakoshh — DISMISSED** [review]

File: docs/guides/external-node/09_decentralization.md (hunk 4 above)
Comment (pompon0) [reply docs/guides/external-node/09_decentralization.md:55]: yes, here I mention only stuff that we want people to use as of today

File: docs/guides/external-node/prepared_configs/mainnet_consensus_config.yaml (hunk 5 above)
Comment (pompon0) [reply docs/guides/external-node/prepared_configs/mainnet_consensus_config.yaml:3]: this is a good question, there are additional constraints on batch that might allow us to make it smaller than just max block size * max blocks per batch. This is a larger topic, let's leave it out of scope of gossipnet announcement.

**Review by dnkolegov — CHANGES_REQUESTED** [review]

File: docs/guides/external-node/09_decentralization.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+> current implementation it may take a couple of hours and gets faster the more nodes you add to the
+> `gossip_static_outbound` list (see below). We are working to remove this inconvenience.
+
+### Generating secrets
+
+Each participant node of the gossipnet has to have an identity (a public/secret key pair) When running your node for the
+first time, generate the secrets be generated by running:
+
+```
+cargo run -p zksync_external_node -- generate-secrets > consensus_secrets.yaml
```
[the remark this answers is not in the document]
Comment (pompon0) [reply docs/guides/external-node/09_decentralization.md:26]: what's the right mask then?

File: docs/guides/external-node/09_decentralization.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+>
+> NEVER reveal the secret keys used by your node. Otherwise someone can impersonate your node on the gossipnet. If you
+> suspect that your secret key has been leaked, you can generate fresh keys using the same tool.
+>
+> If you want someone else to connect to your node, give them your PUBLIC key instead. Both public and secret keys are
+> present in the `consensus_secrets.yaml` (public keys are in comments).
+
+### Preparing configuration file
+
+Copy the template of the consensus configuration file (for [mainnet](perpared_configs/mainnet_consensus_config.yaml) or
```
Comment (RomanBrodetski) [inline docs/guides/external-node/09_decentralization.md:39]: not sure if intended, but the links currently don't work

**Review by RomanBrodetski — APPROVED** [review]

File: docs/guides/external-node/09_decentralization.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (dnkolegov) [reply docs/guides/external-node/09_decentralization.md:26]: 400

[the remark this answers is not in the document]
Comment (pompon0) [reply docs/guides/external-node/09_decentralization.md:26]: done

**Review by dnkolegov — APPROVED** [review]

**Pull request merged**